src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -252,28 +252,35 @@ public int process(List<File> files) throws CheckstyleException {
 
         // Process each file
         for (final File file : files) {
-            if (!CommonUtils.matchesFileExtension(file, fileExtensions)) {
-                continue;
-            }
-            final String fileName = file.getAbsolutePath();
-            fireFileStarted(fileName);
-            final SortedSet<LocalizedMessage> fileMessages = Sets.newTreeSet();
             try {
-                final FileText theText = new FileText(file.getAbsoluteFile(),
-                        charset);
-                for (final FileSetCheck fsc : fileSetChecks) {
-                    fileMessages.addAll(fsc.process(file, theText));
+                if (!CommonUtils.matchesFileExtension(file, fileExtensions)) {
+                    continue;
+                }
+                final String fileName = file.getAbsolutePath();
+                fireFileStarted(fileName);
+                final SortedSet<LocalizedMessage> fileMessages = Sets.newTreeSet();
+                try {
+                    final FileText theText = new FileText(file.getAbsoluteFile(),
+                            charset);
+                    for (final FileSetCheck fsc : fileSetChecks) {
+                        fileMessages.addAll(fsc.process(file, theText));
+                    }
+                }
+                catch (final IOException ioe) {
+                    LOG.debug("IOException occurred.", ioe);
+                    fileMessages.add(new LocalizedMessage(0,
+                            Definitions.CHECKSTYLE_BUNDLE, "general.exception",
+                            new String[] {ioe.getMessage()}, null, getClass(),
+                            null));
                 }
+                fireErrors(fileName, fileMessages);
+                fireFileFinished(fileName);
             }
-            catch (final IOException ioe) {
-                LOG.debug("IOException occurred.", ioe);
-                fileMessages.add(new LocalizedMessage(0,
-                        Definitions.CHECKSTYLE_BUNDLE, "general.exception",
-                        new String[] {ioe.getMessage()}, null, getClass(),
-                        null));
+            catch (Exception ex) {
```

Comment:
yes, I thought about this, but after extra time, It is better to catch Exception there as one day Checked exception will happen (even after some refactoring) in this block and developer will update a method signature but he will miss this catch for sure.

I thought to catch Throwable there as 100% guaranty that on any problem we will report file name ( we could catch OOM Exception on some files eventually), but smth telling me to stay with Exception is ok, and all Errors are problems that most likely do not need filename.